lib/Frontend/CompilerInvocation.cpp

```diff
@@ -45,10 +45,19 @@ static void updateRuntimeLibraryPath(SearchPathOptions &SearchPathOpts,
   llvm::SmallString<128> LibPath(SearchPathOpts.RuntimeResourcePath);
 
   llvm::sys::path::append(LibPath, getPlatformNameForTriple(Triple));
-  SearchPathOpts.RuntimeLibraryPath = LibPath.str();
+  if (Triple.isOSDarwin()) {
+    // On Darwin the library search path is where we store fat binaries.
+    SearchPathOpts.RuntimeLibraryPath = LibPath.str();
+  }
 
   llvm::sys::path::append(LibPath, swift::getMajorArchitectureName(Triple));
   SearchPathOpts.RuntimeLibraryImportPath = LibPath.str();
+  if (!Triple.isOSDarwin()) {
+    // On platforms without fat binaries, we use the arch specific
+    // directory for searching for binaries. This is the same as the import
+    // path for modules.
+    SearchPathOpts.RuntimeLibraryPath = LibPath.str();
+  }
 }
 
```

Comment:
Why not sink the LibPath calculation into the `Triple.isOSDarwin()` or `else` case respectively?

Reply:
Sure. I'll look at writing this be more clear with one branch instead of two. I was over optimizing trying to avoid creating two string allocs for the non-Darwin path at the sake of some clarity. 

Basically:

```
if Darwin:
  RuntimeLibraryPath = /lib/swift/(platform)
else:
  RuntimeLibraryPath = /lib/swift/(platform)/(arch)

RuntimeLibraryImportPath = /lib/swift/(platform)/(arch)
```